src/main/java/org/apache/ibatis/migration/commands/BaseCommand.java

```diff
@@ -315,6 +348,15 @@ protected MigrationLoader getMigrationLoader() {
         : new FileMigrationLoader(paths.getScriptPath(), env.getScriptCharset(), env.getVariables());
   }
 
+  protected Hook createNewMigrationHook() {
+    String before = environment().getBeforeNewHook();
+    String after = environment().getAfterNewHook();
+    if (before == null && after == null) {
+      return NoOpHook.getInstance();
```

Comment:
This follows the [Null Object Pattern](https://en.wikipedia.org/wiki/Null_object_pattern) so that null checks don't need to be carried everywhere.